### PR TITLE
DeleteSAMLProvider AWS

### PR DESCRIPTION
**Pull request by saakovv**
### Summary of the Pull Request

Monitoring DeleteSAMLProvider events separately from UpdateSAMLProvider makes sense, since these actions have different contexts, risks, and consequences. Tracking both types of events allows you to more fully control changes in the configuration of SAML providers and respond faster to potential threats.

### Example Log Event

```
{ [-]
   awsCloud_account_ID: XXXX
   awsRegion: us-east-1
   cwlGroupname: aws-controltower/CloudTrailLogs
   eventCategory: Management
   eventID: 6b09429c
   eventName: DeleteSAMLProvider
   status:success
   eventSource: iam.amazonaws.com
   eventTime: 2024-09-12T07:53:45Z
   eventType: AwsApiCall
   eventVersion: 1.09
   managementEvent: true
   readOnly: false
   recipientAccountId: XXXX
   requestID: f082cd90
   requestParameters: { [-]
     sAMLProviderArn: arn:aws:iam::XXXX:saml-provider/AWSSSO_XXXX_DO_NOT_DELETE
   }
   responseElements: null
   sourceIPAddress: XXXX
   tlsDetails: { [-]
     cipherSuite: TLS_AES_128_GCM_SHA256
     clientProvidedHostHeader: iam.amazonaws.com
     tlsVersion: TLSv1.3
   }
   userAgent: Boto3/1.26.32 Python/3.10.14 Darwin/23.6.0 Botocore/1.29.32
   userIdentity: { [-]
     accessKeyId: XXXX
     accountId: XXXX
     arn: arn:aws:iam::XXXX:user/XXXX
     principalId: XXXX
     type: IAMUser
     userName: XXXX
   }
}
```

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/)

### CODE REVIEW

File: rules/cloud/aws/cloudtrail/aws_delete_saml.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    selection:
+        eventSource: 'iam.amazonaws.com'
+        eventName: 'DeleteSAMLProvider'
+        status: 'success'
+    condition: selection
+falsepositives:
+    - Automated processes using tools like Terraform may trigger this alert.
+    - Legitimate administrative actions by authorized system administrators could cause this alert. Verify the user identity, user agent, and hostname to ensure they are expected.
+    - Deletions by unfamiliar users should be investigated. If the behavior is known and expected, it can be exempted from the rule.
+level: medium
```
Comment (frack113) [inline rules/cloud/aws/cloudtrail/aws_delete_saml.yml:26]: HI,
I am looking for related malicious activity but I don't find it. 
In view of the false positives, it's more of a `low` rule.

**Review by frack113 — CHANGES_REQUESTED** [review]

File: rules/cloud/aws/cloudtrail/aws_delete_saml.yml (hunk 1 above)
Comment (saakovv) [reply rules/cloud/aws/cloudtrail/aws_delete_saml.yml:26]: An attacker can disable SAML, for example, for employees of the information security department. At one of the recent pentests, I saw that the attackers did things that made it difficult for the defense team to work. You simply cannot log into your account via SAML and this is very critical at the time of the attack.

Comment (saakovv) [reply rules/cloud/aws/cloudtrail/aws_delete_saml.yml:26]: Снимок экрана 2024-12-16 в 11 36 52 PM
@frack113 it looks like this

Comment (frack113) [reply rules/cloud/aws/cloudtrail/aws_delete_saml.yml:26]: Thanks for the clarification.

**Review by frack113 — APPROVED** [review]

**Review by nasbench — APPROVED** [review]

**Pull request merged**